firmware/arduino/chessboard/uart.ino

```diff
@@ -3,6 +3,10 @@ char buffer[6];
 int byteNum = -1; // -1 indicates that the start code hasn't been received
 char currentState = '0';
 char errorCode;
+volatile unsigned long previous_activation_time = 0;
+const unsigned long DEBOUNCE_TIME = 100; // Milliseconds
+bool movementFlag = false;
+volatile bool transmitFlag = false;
```

Comment:
Why is one volatile and the other isn't?

Reply:
From my quick reading (and limited understanding) the volatile flag is only supposed to be used for variables that are modified in different threads. So only the two variables that are modified in the interrupt should have volatile flag, which Chase did here

Reply:
I did it wrong. I forgot that serialEvent2() is also a separate thread.

Reply:
Ah kk good catch Lucas